screens/CameraRecording.tsx

```diff
@@ -1,77 +1,178 @@
 import { Camera } from "expo-camera";
-import React, { FunctionComponent, useEffect, useState } from "react";
-import { StyleSheet, View } from "react-native";
+import React, {
+  FunctionComponent,
+  ReactElement,
+  useEffect,
+  useState,
+} from "react";
+import { StyleSheet, View, Image } from "react-native";
 import { Button, Text } from "../components/Themed";
 import { Ionicons } from "@expo/vector-icons";
 import { uploadFileToFirebase } from "../clients/firebaseInteractor";
 
-const CameraRecording: FunctionComponent = () => {
+const SavedOverlay: FunctionComponent<CameraRecordingProps> = ({
+  finished,
+}) => {
+  return (
+    <View style={SavedOverlayViews.container}>
+      <Text style={SavedOverlayViews.header}>
+        Your video diary has been saved
+      </Text>
+      <Text style={SavedOverlayViews.subTitle}>
+        Before you go on to your next recording, please check that your face is
+        visible in the window below.
+      </Text>
+      <Button style={SavedOverlayViews.button} onPress={finished}>
+        <Text style={SavedOverlayViews.buttonText}>Continue</Text>
+      </Button>
+    </View>
+  );
+};
+
+const SavedOverlayViews = StyleSheet.create({
+  container: {
+    position: "absolute",
+    left: 0,
+    top: 0,
+    width: "100%",
+    height: "100%",
+    backgroundColor: "rgba(0, 0, 0, 0.8)",
+    justifyContent: "center",
+  },
+  header: {
+    color: "white",
+    fontWeight: "500",
+    fontSize: 30,
+    margin: "5%",
+  },
+  subTitle: {
+    color: "white",
+    fontWeight: "400",
+    fontSize: 18,
+    marginHorizontal: "5%",
+  },
+  button: {
+    position: "absolute",
+    width: "90%",
+    bottom: "10%",
+    left: "5%",
+    backgroundColor: "#0038FF",
+  },
+  buttonText: {
+    color: "white",
+    fontSize: 20,
+    textAlign: "center",
+    padding: 20,
+  },
+});
+
+const COUNTDOWN_TIME = 5;
+
+export interface CameraRecordingProps {
+  overlay: ReactElement;
+  finished: () => void;
+}
+
+const CameraRecording: FunctionComponent<CameraRecordingProps> = ({
+  overlay,
+  finished,
+}) => {
   const [recording, setRecording] = useState(false);
-  const [permissions, setHasPermission] = useState(false);
   const [camera, setCamera] = useState<Camera | null>(null);
   const [time, setTime] = useState(0);
+  const [countdownTime, setCountdownTime] = useState(COUNTDOWN_TIME);
   useEffect(() => {
-    Camera.requestPermissionsAsync().then((response) => {
-      setHasPermission(response.status === "granted");
-    });
-  }, []);
-  useEffect(() => {
+    let unsubscribe: number | undefined = undefined;
     if (recording) {
-      window.setTimeout(() => setTime(time + 1000), 1000);
+      unsubscribe = window.setTimeout(() => setTime(time + 1000), 1000);
     }
+    return () => window.clearTimeout(unsubscribe);
   }, [time, recording]);
-  if (!permissions) {
-    return <Text>Please give permissions</Text>;
```

Comment:
Do we still need to worry about permissions?

Reply:
Not for this screen, since the `PrepareFace` has that baby checked for us.